index.js

```diff
@@ -31,45 +37,116 @@ client.on('messageCreate', async message => {
                     return; //This will break out of the loop and function, might not be the best approach, but it's the one we can take here.
                 }
+
                 const commandInput = match[1];
                 //console.log(`Command Input: "${commandInput}"`);
-                //Perform a better regex check here for data sanitization on searchTerm
+                //Perform a regex check here for data sanitization on commandInput
                 //  - a-zA-Z (),|-!' - (space is included)
+                const commandMatches = commandInput.match(/^([aAtTiIvV]\:)?([a-zA-Z \(\),\|\-!\']+)$/);
+
+                //console.log(`Matches ${commandMatches}`);
+
+                if(commandMatches){
+                    //console.log(`commandMatches[1]: "${commandMatches[1]}"`);
+
+                    var defaultView = commandMatches[1] === undefined || commandMatches[1] === '';
+                    var textView = commandMatches[1] && commandMatches[1].toLowerCase() === "t:";
+                    var imageView = commandMatches[1] && commandMatches[1].toLowerCase() === "i:";
+                    var verboseView = commandMatches[1] && commandMatches[1].toLowerCase() === "v:";
+
+                    //[[card]] [[t:card]] [[i:card]] [[v:card]]
+                    var cardSearch = defaultView || textView || imageView || verboseView;
 
-                //Perform split on : if possible?
-
-                //[[card]] should show the default view
-                //[[t:card]] should show the text, no image
-                //[[i:card]] should only the image, nothing else
-                //[[v:card]] should show verbose mode (all the things)
-                //[a:ping], [a:uptime], [a:stats] -- administrative functions
-                //[a:days] -- days since the kickstarter for HR successfully funded
-                var cardSearch = true;
-                //"Default View"
-                var showText = false;
-                var showImage = true;
-                var showType = true;
-                var showCost = false;
-                var showFaction = false;
-                var showDefense = false;
-
-                //We figured out the user wanted a card search, so let's get the results!
-                if(cardSearch){
-                    //figure out what the searchTerm is (after changing the above to commandInput)
-                    const searchTerm = commandInput
-                    const results = await this.gallery.searchGallery(searchTerm, MAX_RESULTS);
-                    if (results.length === 0) {
-                        await message.reply(`No cards found with name "${searchTerm}"`);
-                        continue;
+                    //[[a:???]] -- administrative functions
+                    var adminCommand = commandMatches[1] && commandMatches[1].toLowerCase() === 'a:';
```

Comment:
This can be `const` as well.

```suggestion
                    const adminCommand = commandMatches[1] && commandMatches[1].toLowerCase() === 'a:';
```